src/sentry/incidents/serializers/alert_rule_trigger_action.py

```diff
@@ -144,6 +149,32 @@ def validate(self, attrs):
                         {"sentry_app": "The installation does not exist."}
                     )
 
+        if attrs.get("priority"):
+            if action_type not in [
+                AlertRuleTriggerAction.Type.PAGERDUTY,
+                AlertRuleTriggerAction.Type.OPSGENIE,
+            ]:
+                raise serializers.ValidationError(
+                    {"priority": "Can only be set for Pagerduty or Opsgenie"}
+                )
+
+            priority: str = attrs["priority"]
```

Comment:
nit: might be good to use a const, since we're using/relying on this key through the code base